### PR TITLE
Update sqs index add region code filtering

### PR DESCRIPTION
**Pull request by alexgleith**
This adds a new CLI argument that can be a file path or URL that should contain a list of keys that are checked against the 'parameters'/'odc:region_code' field and this can be used as a filter.

Tested locally.

### CODE REVIEW

File: odc_index/sqs_to_dc.py [hunk 1]
```diff
@@ -235,12 +198,69 @@ def do_indexing(
...
+    **kwargs,
+) -> Tuple[int, int]:
+
+    ds_success = 0
+    ds_failed = 0
+
+    doc2ds = Doc2Dataset(dc.index, products=products, **kwargs)
+
+    if region_code_list_uri:
+        region_codes = pd.read_csv(region_code_list_uri)
```
Comment (Kirill888) [inline odc_index/sqs_to_dc.py:224]: Turning it into a set would probably be better

```python
region_codes = pd.read_csv(region_code_list_uri)
region_codes = set(region_code.values.ravel())
```

File: odc_index/sqs_to_dc.py [hunk 2]
```diff
@@ -235,12 +198,69 @@ def do_indexing(
...
+    odc_metadata_link=False,
+    region_code_list_uri=None,
+    **kwargs,
+) -> Tuple[int, int]:
+
+    ds_success = 0
+    ds_failed = 0
+
+    doc2ds = Doc2Dataset(dc.index, products=products, **kwargs)
+
```
Comment (Kirill888) [inline odc_index/sqs_to_dc.py:222]: set `region_codes = None`, best to not have "some variables are defined <only if something>"

**Review by pindge — APPROVED** [review]

File: odc_index/sqs_to_dc.py [hunk 3]
```diff
@@ -235,12 +200,73 @@ def do_indexing(
...
+                if not record_path:
+                    # Extract metadata and URI for indexing
+                    metadata, uri = get_metadata_uri(
+                        metadata, transform, odc_metadata_link
+                    )
+                else:
+                    metadata, uri = get_metadata_from_s3_record(metadata, record_path)
+
+                # If we have a region_code filter, do it here
+                if region_code_list_uri:
```
Comment (sachitrajbhandari) [inline odc_index/sqs_to_dc.py:248]: Can we check for `region_codes` instead of `region_code_list_uri`?

**Review by sachitrajbhandari — APPROVED** [review]

File: odc_index/sqs_to_dc.py (hunk 3 above)
Comment (alexgleith) [reply odc_index/sqs_to_dc.py:248]: I already merged this...

I did wonder if I should check by the `region_code_list_uri` and `region_codes` ... which is probably better. 

If there's no items in the file that's retrieved this code will fail. But maybe it should?

**Pull request merged**